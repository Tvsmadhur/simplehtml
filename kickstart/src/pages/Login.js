import React, { useState, useEffect } from 'react'

import { login } from '../api/Services';
import { ToastContainer, toast } from 'react-toastify';
import { useMutation, useQueryClient } from 'react-query';
import { useNavigate } from "react-router-dom";


function Login() {
  const navigate = useNavigate();
  const queryClient = useQueryClient();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [userError, setUserError] = useState("");
  const [usernameerror, setUsernameerror] = useState("");
  const [passworderror, setPassworderror] = useState("")
  const { mutate, isLoading } = useMutation(login, {
    onSuccess: response => {
      console.log(response);
      toast.success('Login Succesfull');
      localStorage.setItem("token", response.token)
      navigate("/users")
    },
    onError: () => {
      toast.error('Login Failed');
    }
  })
  const changeHandler = (e) => {
    console.log("firing")
    setPassworderror("");
    setUserError("");
    setUsernameerror("")
    if (e.target.name == "username") {
      setUsername(e.target.value)
    }
    else if (e.target.name == "password") {

      setPassword(e.target.value)
    }
  }

  const formHandler = () => {
    console.log("firing")
    if (username == "" && password == "") {
      setUserError('All fields are required')
    }
    else if (username == "") {
      setUsernameerror("Username is required")
    }
    else if (password == "") {
      setPassworderror("Password is required");
    }
    else {
      const data = {
        email: username,
        password: password
      }

      // login(data).then((response) => {
      //   console.log("response", response)
      //   toast.success('Login Succesfull');
      //   localStorage.setItem("token", response.token)
      //   navigate("/users")
      // }).catch((error) => {
      //   console.log("error", error)
      //   toast.error('Login Failed');
      // })

      mutate(data)

    }
  }

  return (
    <>
      <section className="Form my-4 mx-5">
        <div className="container no-gutters">
          <div className="row no-gutters">
            <div className="col-lg-5">
              <img role="img" src="https://images.unsplash.com/photo-1664575601786-b00156752b61?ixlib=rb-4.0.3&ixid=MnwxMjA3fDF8MHxzZWFyY2h8MXx8d29ya3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=600&q=60" className="img-fluid" alt="image"></img>
            </div>
            <div className="col-lg-7 px-5 pt-5" >
              <h4 data-testid="heading">Login</h4>
              <form>
                <div className="form-row mb-4">
                  <div className="col-lg-7">
                    <input role="input" name="username" data-testid="username" type="email" placeholder="Enter email" onChange={(e) => changeHandler(e)} className="form-control my-3 p-2" />
                    <p role="error" style={{ color: "red" }}>{usernameerror}</p>
                  </div>
                </div>

                <div className="form-row mb-4">
                  <div className="col-lg-7">
                    <input role="input" name="password" type="password" data-testid="password" placeholder="********" onChange={(e) => changeHandler(e)} className="form-control my-3 p-2 " />
                    <p role="error" style={{ color: "red" }}>{passworderror}</p>

                  </div>
                </div>

                <div className="form-row">
                  <div className="col-lg-7">
                    <button type="button" data-testid="button" onClick={formHandler} className="btn1 mt-3 mb-5">Login</button>
                  </div>
                </div>
                {/* <a href="#">Forgot password</a> */}
                <p role="error" style={{ color: "red" }}>{userError}</p>
                <p>Don't have an account? <a href="/signup">Register here</a></p>

              </form>
            </div>


          </div>
        </div>
      </section>

    </>
  )
}
export default Login