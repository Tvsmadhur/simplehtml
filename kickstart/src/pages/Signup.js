import React, { useState, useEffect } from 'react'

import { ToastContainer, toast } from 'react-toastify';
import { useNavigate } from "react-router-dom";
import { useMutation, useQueryClient } from 'react-query';
import { signup } from '../api/Services';

function Signup() {
  const navigate = useNavigate();
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [userError, setUserError] = useState("");
  const [usernameerror, setUsernameerror] = useState("");
  const [passworderror, setPassworderror] = useState("");
  const { mutate, isLoading } = useMutation(signup, {
    onSuccess: response => {
      console.log(response);
      toast.success('Registration Succesfull');
      localStorage.setItem("token", response.token)
      navigate("/users")
    },
    onError: () => {
      toast.error('Registration failed');
    }
  })

  const changeHandler = (e) => {
    setPassworderror("");
    setUserError("");
    setUsernameerror("")
    if (e.target.name == "username") {
      setUsername(e.target.value)
    }
    else if (e.target.name == "password") {
      setPassword(e.target.value)
    }
  }
  const formHandler = () => {
    if (username == "" && password == "") {
      setUserError('All fields are required')
    }
    else if (username == "") {
      setUsernameerror("Username is required")
    }
    else if (password == "") {
      setPassworderror("Password is required");
    }
    else {
      const data = {
        email: username,
        password: password
      }
      // signup(data).then((response) => {
      //   console.log("response", response)
      //   toast.success('Registration Succesfull');
      //   localStorage.setItem("token", response.token)
      //   navigate("/users")
      // }).catch((error) => {
      //   console.log("error", error)
      //   toast.error('Registration Failed');
      // })
      mutate(data)
    }

  }
  return (
    <>
      <section className="Form my-4 mx-5">
        <a href="/"> <button className="btn2" data-testid="back">Back</button></a>
        <div className="container no-gutters">
          <div className="row no-gutters">
            <div className="col-lg-5">
              <img src="https://images.unsplash.com/photo-1664575601786-b00156752b61?ixlib=rb-4.0.3&ixid=MnwxMjA3fDF8MHxzZWFyY2h8MXx8d29ya3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=600&q=60" className="img-fluid" alt="image"></img>
            </div>
            <div className="col-lg-7 px-5 pt-5" >
              <h4>Register</h4>
              <form>
                <div className="form-row mb-4">
                  <div className="col-lg-7">
                    <input name="username" type="email" placeholder="Enter email" onChange={changeHandler} className="form-control my-3 p-2" />
                    <p style={{ color: "red" }}>{usernameerror}</p>
                  </div>
                </div>

                <div className="form-row mb-4">
                  <div className="col-lg-7">
                    <input name="password" type="password" placeholder="Password" onChange={changeHandler} className="form-control my-3 p-2 " />

                    <p style={{ color: "red" }}>{passworderror}</p>
                  </div>
                </div>
                <div className="form-row">
                  <div className="col-lg-7">
                    <button type="button" className="btn1 mt-3 mb-5" onClick={formHandler}>Register</button>
                  </div>
                </div>
                <p style={{ color: "red" }}>{userError}</p>


              </form>
            </div>


          </div>
        </div>
      </section>

    </>
  )
}
export default Signup