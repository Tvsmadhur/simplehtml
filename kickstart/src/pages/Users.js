import React, { useState, useEffect } from 'react'

import { getUsers } from '../api/Services';
import { useNavigate } from "react-router-dom";
import { useQuery } from 'react-query';
import { ToastContainer, toast } from 'react-toastify';
function Users() {
  const [users, setUsers] = useState()

  const navigate = useNavigate();
  const { data, isError, isLoading, isFetching } = useQuery('users', getUsers);
  useEffect(() => {
    // get("https://reqres.in/api/users?page=2").then((response) => {
    //   setUsers(response.data.data)
    // })
    //   .catch((error) => {
    //     console.log(error)
    //   })
    // fetchData()

  }, [])
  // const fetchData = () => {
  //   getUsers().then((response) => {

  //     setUsers(response.data)
  //   }).catch((error) => {
  //     console.log(error)
  //   })
  // }

  const logout = () => {

    localStorage.clear()
    navigate("/")
  }
  if (isError) {
    return toast.error('Fetching data  Failed');
  }
  console.log({ isFetching })

  return (
    <>
      <button data-testid="button-users" className="btn3 float-right" onClick={logout}>Logout</button>

      {typeof data != "undefined" ?
        <div className="d-flex justify-content-center container flex-wrap">
          {data.data.map((item, index) => {
            return [
              <>

                <div className="card" onClick={() => {
                  navigate(`/user/${item.id}`);
                }} key={index} style={{ width: "18rem" }}>
                  <img className="card-img-top" src={item.avatar} alt="Card image cap" />
                  <div className="card-body">
                    <h4 className="card-text">Name:{item.first_name}{item.last_name}</h4>
                    <p className="card-text">Email:{item.email}</p>
                  </div>
                </div>
              </>
            ]
          })

          }
        </div> : <div>Loading....</div>}

    </>
  )
}
export default Users