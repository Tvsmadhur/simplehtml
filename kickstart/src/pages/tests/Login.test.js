import Login from '../Login'
import Users from '../Users';
import Signup from '../Signup';
import { render, cleanup, screen, fireEvent, getByRole, userEvent, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';

import * as router from 'react-router'

import "@testing-library/jest-dom";
import { createBrowserHistory } from 'history'
import { createMemoryHistory } from 'history';



import { BrowserRouter, Switch, Router, Routes, Route, MemoryRouter, useNavigate } from 'react-router-dom'

import {

  QueryClient,
  QueryClientProvider,
} from 'react-query'
const queryClient = new QueryClient()
// const history = createBrowserHistory()


const LoginComponent = (component) => {

  const history = createMemoryHistory();

  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter history={history}>
        <Routes>
          <Route exact path={"/"}
            element={component} />
          <Route exact path={"/users"}
            element={<Users />} />
        </Routes>
      </BrowserRouter>
    </QueryClientProvider>
  )
}

afterEach(cleanup);


const LoginScreen = () => {
  const history = createMemoryHistory();
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter history={history}>

        <Login />
      </BrowserRouter>
    </QueryClientProvider>
  )
}



describe('checking Every component is rendered correctly', () => {


  it('should take snapshot', async () => {
    const { container } = render(<LoginScreen />);
    expect(container).toMatchSnapshot()

  })
  it('should render all the documents', async () => {
    const { getByText, getByTestId, asFragment, getAllByRole, } = render(LoginComponent(<Login />));
    const headNode = getByTestId("heading")
    const inputNode = getAllByRole('input')
    expect(getByTestId('heading')).toHaveTextContent("Login")
    expect(screen.getByRole('img')).toBeInTheDocument();
    const primaryButton = screen.getByRole('button', { name: /Login/i })
    expect(getByTestId('heading')).toHaveTextContent("Login")
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(primaryButton).toBeInTheDocument()
    expect(inputNode).toHaveLength(2)
  })
  it('to Check button event to be triggered', async () => {
    const history = createMemoryHistory();
    history.push("/signup")



    const { getByText, getByTestId, asFragment, getAllByRole, getByRole, debug } = render(
      <QueryClientProvider client={queryClient}>
        <BrowserRouter history={history}>
          <Routes>
            <Route path="/" component={<Login />} />
          </Routes>
        </BrowserRouter>
      </QueryClientProvider>

    );
    const logSpy = jest.spyOn(console, "log");
    // const button = getByTestId('button')


    // expect(screen.getAllByRole('error')[0]).toBeInTheDocument()
    // expect(screen.getAllByRole('error')[1]).toBeInTheDocument()
    // expect(screen.getAllByRole('error')[2]).toBeInTheDocument()

    // fireEvent.click(button);
    // expect(logSpy).toHaveBeenCalledTimes(1);
    // history.push("/signup")
    expect(history.location.pathname).toBe("/signup")
    // expect(screen.getByTestId('back')).toBeInTheDocument()


    debug()
  })


  it('to Check onchange event to be triggered', () => {
    const { getByText, getByTestId, asFragment, getAllByRole } = render(<LoginScreen />);
    const logSpy = jest.spyOn(console, "log");
    const formHandler = jest.fn();
    const inputNode = getByTestId('username')

    fireEvent.change(inputNode, { target: { value: 'new value' } })
    expect(logSpy).toHaveBeenCalled();
    const inputPasswordNode = getByTestId('password')
    fireEvent.change(inputPasswordNode, { target: { value: 'new value' } })
    expect(logSpy).toHaveBeenCalled();

  })

})