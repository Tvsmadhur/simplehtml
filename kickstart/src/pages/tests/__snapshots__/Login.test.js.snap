// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking Every component is rendered correctly should take snapshot 1`] = `
<div>
  <section
    class="Form my-4 mx-5"
  >
    <div
      class="container no-gutters"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="col-lg-5"
        >
          <img
            alt="image"
            class="img-fluid"
            role="img"
            src="https://images.unsplash.com/photo-1664575601786-b00156752b61?ixlib=rb-4.0.3&ixid=MnwxMjA3fDF8MHxzZWFyY2h8MXx8d29ya3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=600&q=60"
          />
        </div>
        <div
          class="col-lg-7 px-5 pt-5"
        >
          <h4
            data-testid="heading"
          >
            Login
          </h4>
          <form>
            <div
              class="form-row mb-4"
            >
              <div
                class="col-lg-7"
              >
                <input
                  class="form-control my-3 p-2"
                  data-testid="username"
                  name="username"
                  placeholder="Enter email"
                  role="input"
                  type="email"
                />
                <p
                  role="error"
                  style="color: red;"
                />
              </div>
            </div>
            <div
              class="form-row mb-4"
            >
              <div
                class="col-lg-7"
              >
                <input
                  class="form-control my-3 p-2 "
                  data-testid="password"
                  name="password"
                  placeholder="********"
                  role="input"
                  type="password"
                />
                <p
                  role="error"
                  style="color: red;"
                />
              </div>
            </div>
            <div
              class="form-row"
            >
              <div
                class="col-lg-7"
              >
                <button
                  class="btn1 mt-3 mb-5"
                  data-testid="button"
                  type="button"
                >
                  Login
                </button>
              </div>
            </div>
            <p
              role="error"
              style="color: red;"
            />
            <p>
              Don't have an account? 
              <a
                href="/signup"
              >
                Register here
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
